cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -2,61 +2,92 @@ import { getLogger } from '@raincatcher/logger';
 
   // tslint:disable-next-line:max-line-length
-  constructor(protected readonly userRepo: UserRepository, protected readonly userService: UserService, loginRoute?: string) {
-    this.loginRoute = loginRoute || '/login';
+  constructor(protected readonly userRepo: UserRepository, protected readonly userService: UserService) {
   }
 
   /**
-   * Initializes an Express application to use passport and express-session
+   * Initializes an Express application to use passport.
+   * Note: If the session options are not defined, session based authentication will not be used.
    *
    * @param app - An express application
    * @param sessionOpts - Session options to be used by express-session
+   * @param secret - Secret to be used for Passport's JWT strategy
    */
-  public init(app: express.Express, sessionOpts: SessionOptions) {
-    getLogger().info('Initializing express app to use express session and passport');
-    app.use(session(sessionOpts));
-    app.use(passport.initialize());
-    app.use(passport.session());
-    this.setup(passport);
+  public init(app: express.Router, sessionOpts?: SessionOptions, secret?: any) {
+    getLogger().info('Initializing express app to use passport');
+    if (sessionOpts) {
+      app.use(session(sessionOpts));
+      app.use(this.passport.initialize());
+      app.use(this.passport.session());
+      this.setupCookie(this.passport);
+    } else {
+      this.jwtOpts = {
+        jwtFromRequest: ExtractJwt.fromAuthHeader(),
+        secretOrKey: secret || CONSTANTS.defaultSecret
+      };
+      app.use(this.passport.initialize());
+      this.setupToken(this.passport, this.jwtOpts);
+    }
   }
 
   /**
    * Function which checks if the user requesting access to the resource is authenticated and authorized to
-   * access the resource. Redirects to the login page if user is not authenticated or returns a status of 403
+   * access the resource. Returns a status of 401 if the user is not authenticated and returns a status of 403
    * if the user does not have the required role.
    *
    * @param role - Role which the user needs in order to access this resource
    */
   public protect(role?: string) {
     const self = this;
     return (req: express.Request, res: express.Response, next: express.NextFunction) => {
-      if (!req.isAuthenticated()) {
-        if (req.session) {
-          // Used for redirecting to after a successful login when option successReturnToOrRedirect is defined.
-          req.session.returnTo = self.setReturnToUrl(req);
-          req.session.clientURL = req.session.returnTo;
+      let hasRole;
+
+      if (req.headers && req.headers.authorization) {
+        getLogger().info('Token based authentication and authorization');
+        const token = req.headers.authorization.toString().substring(4);
+        try {
+          const user = jwt.verify(token, this.jwtOpts.secretOrKey);
+          hasRole = role ? this.userService.hasResourceRole(user, role) : true;
```

Comment:
IMHO `role ? A : B` makes this code unreadable - try to avoid that in the future, especially in mission critical code like security implementation.